crates/nu-cli/src/repl.rs

```diff
@@ -114,6 +115,10 @@ pub fn evaluate_repl(
         engine_state.merge_env(&mut unique_stack, cwd)?;
     }
 
+    if shell_integration {
+        do_shell_integration(System::host_name(), engine_state, &mut unique_stack);
```

Comment:
1. I'd rather get this host_name() once and use pass it through. Seems like that minor refactor would help.
2. I also think `do_shell_integration()` should be named something else since it's only part of the shell integration. Not you're fault, of course, but you are renaming it here. Maybe `shell_integration_osc_7_633_2()` which leads us to another PR which could separate those.

I'm up for other names.

Reply:
Both good comments that address things that were bugging me too whilst I was writing this PR — I'll have a quick think about if any other name would be better, but otherwise I'll make these changes ASAP!